parsing.py

```diff
@@ -100,8 +100,11 @@ def parse_usi(usi: str) -> Tuple[sus.MsmsSpectrum, str, str]:
             raise e
     try:
         collection = match.group(1).lower()
-        # Send all proteomics USIs to MassIVE.
-        if (collection.startswith('msv') or collection.startswith('pxd') or
+        annotation = match.group(5)
+        # Send all proteomics USIs (by definition all annotated USIs) to MassIVE.
+        # mzdraft USIs are assumed to also use ProForma notation.  If this changes,
+        # be sure to change this logic.
+        if (annotation is not None or collection.startswith('msv') or collection.startswith('pxd') or
```

Comment:
Is the problem here the difference between GNPS tasks and MassIVE tasks? And MassIVE tasks can't be resolved by GNPS and vice versa?

I'm not really convinced of this solution.

1. This will not fix the problem if people mix up the collection but don't specify an annotation.
2. Disregarding MassIVE vs GNPS tasks, currently all annotated USIs should be proteomics ones and will be sent to MassIVE based on their collection specifier anyway.
3. According to the USI standard, the annotation can be a random string. A completely valid small molecule could have some annotation if people want to include more information with it, even though it will currently not be parsed to enrich the generated plot. But it's still a pretty reasonable situation. This change will cause such USIs to fail.

So I'm not sure that this is the best solution for the issue of conflicting MassIVE vs GNPS tasks. The potential downsides are bigger, and we introduce technical debt with this hacky fix. Instead, maybe a relevant error message for this edge case would be more useful?

Reply:
The main issue here is that if a MassIVE task is specified with a GNPS identifier and has a ProForma annotation, it is parsed improperly - when the regex was incorrect this caused errors, but with the fixed regex all that would happen is the annotation not be considered.

1. All MassIVE and GNPS tasks can be resolved by MassIVE as I'm aware (and likely the other way around too).  So I think without an annotation this isn't super critical.
2. Correct, but in theory any MassIVE task _can_ have a GNPS identifier and visa versa (since it's all in the same system).  I don't see this going away anytime soon.  But even if they were split, GNPS tasks might have peptide identifiers right? I haven't seen examples of cyclopeptides but aren't some in GNPS (not that we necessarily would annotate them very well, given all the internal fragmentation to consider)?
3. My wording was too strong here, at the moment I think this is true, but yes, according to the spec it could be different.

For sure this is a short term gain vs potential long term issue - and I of course defer to you on this.  My feeling is to just to piggy-back on MassIVE ProForma resolution until there is a reason to do otherwise.

Reply:
Ok, I think I'm starting to understand the issue better. Would a solution be to strip the annotation before submitting to GNPS? Does that get rid of the parsing error?

I'm adding ProForma support to spectrum_utils, so we can offload parsing to annotation to there. This would then also natively enable ProForma support for any backend.
The blocking issue was that the Unimod OBO was invalid, but I've just got confirmation from John Cottrell today that they've updated their OBO with fixes I proposed. I just need to make some small updates with the valid OBO and add unit tests to include (pretty complete) ProForma support in spectrum_utils.

Reply:
Sending the USI with the annotation to GNPS wouldn't be an error anymore (with the fix on line 53 of parsing.py), but it just wouldn't return the parsed annotation at all.  So if you did indeed want a spectrum annotated with a peptide and just happened to write a task with GNPS instead of MassIVE in the USI, it would not be annotated.

Keeping this separation between MassIVE and GNPS tasks and adding native ProForma would also solve the problem (though of course it would be best to ensure that the ProForma parser is consistent between the two, so if I pass a GNPS task with a peptide annotation it would return the same peak annotations as if I pass the the same USI with MassIVE instead of GNPS.)

Do keep in mind that this PR is in response to this issue https://github.com/mwang87/MetabolomicsSpectrumResolver/issues/158 which indicates that somehow the behavior of a GNPS task with a ProForma ID should be handled correctly in the current code, and this PR fixes this codebase so that it correctly parses the USI to know how to pass it on to a service that can handle them.

It sounds like a fine idea to revert this PR, given what you've mentioned.  It might be nice to wait until the ProForma support is fully implemented, but even before it should be fine too, there will just be a space in which issue #158 becomes unresolved again.

Reply:
All right, thanks for the info and figuring out this issue. I discussed it with Ming a bit more yesterday, and we'll keep this PR in for now to avoid the problem, but we'll indeed revisit it with native ProForma resolution a bit later.